tgui/packages/tgui/interfaces/ForensicScanner.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  Box,
+  Button,
+  Icon,
+  LabeledList,
+  NoticeBox,
+  Section,
+} from 'tgui-core/components';
+import { capitalizeFirst } from 'tgui-core/string';
+
+import { useBackend } from '../backend';
+import { Window } from '../layouts';
+
+export type ForensicScannerData = {
+  log_data: LogData[];
+};
+
+type LogData = {
+  scan_target: string;
+  scan_time: string;
+  Prints: Record<string, string>;
+  Fibers: Record<string, string>;
+  Blood: Record<string, string>;
+  Reagents: Record<string, number>;
+  'ID Access': Record<string, string[]>;
+};
+
+export const ForensicScanner = (props) => {
+  const { act, data } = useBackend<ForensicScannerData>();
+  const { log_data = [] } = data;
+  return (
+    <Window width={512} height={512}>
+      <Window.Content>
+        {log_data.length === 0 ? (
+          <NoticeBox>Log empty.</NoticeBox>
+        ) : (
+          <Section
+            title="Scan history"
+            fill
+            scrollable
+            buttons={
+              <>
+                <Button.Confirm
+                  icon="trash"
+                  color="danger"
+                  onClick={() => act('clear')}
+                >
+                  Clear logs
+                </Button.Confirm>
+                <Button icon="print" onClick={() => act('print')}>
+                  Print report
+                </Button>
+              </>
+            }
+          >
+            {log_data.reverse().map((log) => (
+              <Section
+                key={log.scan_time}
+                title={`${capitalizeFirst(log.scan_target)} scan at ${log.scan_time}`}
+              >
```

Comment:
does using duplicate keys throw an error? theres a small chance that you could fuck over your UI logs if you happen to scan two items at the same station time. if it does, you should find another key, create a random one at the time the log gets created or use the stringified index (call reverse after map in that case), last one should be safe too since `log_data` wont change the contents, it only gets appended to or cleared

Reply:
Scanning has a 3 second cooldown.

Reply:
`station_time_timestamp()` loops back around